### ISSUE TITLE
`rectangleBorder` broken

### ISSUE THREAD

**Issue by na2axl** [aui-framework/aui#350]
Hey @Alex2772,

I tried to use `rectangleBorder` from the `IRenderer` interface for a custom view, but the result is notz drawn properly, finally had to use `roundedRectangleBorder` with a radius of 0 to have the correct result. There is the current output:
![image](https://github.com/user-attachments/assets/6f78e403-0d49-47cd-8575-acfd51e394f3)

The semi-rectangle in red is the result of `rectangleBorder`, and the rectangle in blue is the result of `roundedRectangleBorder`.

```cpp
    // Outer box (wrong result)
    ctx.render.rectangleBorder(
      ASolidBrush { colors::secondary._600 },
      outer_offset,
      outer_size,
      2_dp);

    // Inner box (correct result)
    ctx.render.roundedRectangleBorder(
      ASolidBrush { colors::primary._600 },
      inner_offset,
      inner_size,
      0,
      2_dp);
```

**Comment by Alex2772** [comment]
Yes it’s known issue. Yes, using rounded rectangle border is a valid workaround. [We use it as well](https://github.com/aui-framework/aui/blob/a4a2fd7aea520e5eaf3548a720ad45f5e4e10e2b/aui.views/src/AUI/ASS/Property/Border.cpp#L30).

Thank you for rising this issue.

I’ll keep this issue open until further investigation.
